src/Telemetry/Library/Telemetry.csproj

```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>

```

Comment:
I'm not certain of how these frameworks work, but most everything else in the compiler repo is netcoreapp3.1 or netstandard2.1. Is it going to create compatibility issues for this to be targeting net5.0?

Reply:
net5.0 supercedes netcoreapp3.1 and netstandard2.1 both; in particular, a net5.0 exe or dll can reference a netstandard2.1 project. The bad case is referring to a net5.0 dll from a netcoreapp3.1 exe. That said, we're in the process of upgrading to net6.0 for exes across the board, such that this should work well once bumped to net6.0.

Reply:
That sounds especially problematic for what the intended use of this library is. This library needs to be referenced by other projects in the repo. It sounds like this needs to be lowered to at least netcoreapp3.1, if not netstandard2.1.

Reply:
@cgranade, the initial use case for this library is for the Q# formatter (which targets netstandard2.1) to use it.
Do you foresee a problem that would require us to temporarily target netstandard2.1, maybe by having different build profiles?

Reply:
just saw your comment, @ScottCarda-MS. I'll work on an option to target the older frameworks.

Reply:
@ScottCarda-MS: Given that those exe projects will be upgraded to net6.0 ASAP (.NET 6.0 launches and is the default early this month), I don't think it's problematic at all for this library to target net6.0. There's not much reason to upgrade existing libraries from netstandard2.1 to net6.0, but I also think there's also not a lot of reason to develop _new_ libraries against netstandard2.1 instead of net6.0. In particular, the net6.0 TFM brings with it support for C# 10 (netstandard2.1 targets 7.0, IIRC); that in turn makes it easier to provide null-safe code, given improvements to nullable reference types in more recent versions of C#.

+@ricardo-espinoza for .NET 6.0 discussion/impact.

Reply:
@cgranade If the update to net6.0 is, say this week, than we can hold off on this PR till then. Otherwise we are in a bit of a time crunch to get this in, and we need to have this PR free of dependencies on code changes that will happen later in the month. In general code should be written to be compatible with the codebase as it is, not as it will be. Based on that, I think it will be more appropriate to updated this library to net6.0 as well as the Q# formatter along with everything else later in the month.

Reply:
I'll let @ricardo-espinoza speak in more detail to specific timelines, but given that .NET 6.0 goes live in a few days, I do believe the update to the net6.0 TFM is intended to be somewhat timely.

> In general code should be written to be compatible with the codebase as it is, not as it will be.

It may be helpful to assume good faith on the part of other developers! In particular, I think it's relevant in this case that good code should also anticipate future needs, especially when those needs are planned on a short time frame. Considering how code is affected by other features and maintenance can be a good way of preventing churn.

For example, one solution to the concern that you raised would be to temporarily enable `netstandard2.1` support as per @vxfield's comment, but to do so with multi-framework targeting:

```xml
<TargetFrameworks>net6.0;netstandard2.1</TargetFrameworks>
```

That ensures that dependencies can be taken now as-is, and also that code builds correctly against changes currently in progress.

Reply:
@ScottCarda-MS, any chance you can make the Q# Formatter target .NET 5 as of now?
Downgrading the Telemetry library back to netstandard2.1 will require a lot of (silly) code changes from C# 9 to C# 8 that I would like to avoid if possible.
Let me know.

Reply:
Yah I can look into that if it will save you a lot of changes like that. I'll need to double check with Bettina if that will work with the SDK integration she is working on, but I expect that it will be okay to go forward with that. For now let's assume we can, and I'll update the Q# Formatter.